### DISCUSSION TITLE
Error uploading to print host: Operation timed out

### DISCUSSION THREAD

**Discussion by BrentEBeebe** [OrcaSlicer/OrcaSlicer#5852] [Q&A]
I'm new to OrcaSlicer, trying to get it to print a loaded design.

When I try toUpload and print, I get "Error uploading to print host: Operation timed out".

But if I test the connection, it says "Connection to Flashforge works correctly".

Screen captures below.  Suggestions appreciated.  Looks like a great tool and I look forward to using it versus the "came with the printer" tools I have now.
<img width="737" alt="Screen Shot 2024-06-24 at 5 07 58 PM" src="https://github.com/SoftFever/OrcaSlicer/assets/173743474/8ac6e5c8-8bd1-4002-88f7-571e8a465764">
<img width="617" alt="Screen Shot 2024-06-24 at 5 07 05 PM" src="https://github.com/SoftFever/OrcaSlicer/assets/173743474/ccd6f231-ce9e-4128-a98f-5220a25114ad">

**Comment by DarrylBarney** [comment]
I have the same issue - same error message when uploading, same result - Printer is also Flashforge Adventurer 3

**Comment by ThomasB-G** [comment]
Same issue, works with test connection but after a few seconds while its uploading it just stops and gives this error. Printer is a Flashforge Adventurer 5m

**Reply by RyantHults** [reply]
I am seeing the same thing on my flashforge A5M. It looks like if the file is small enough to finish before the timeout is hit, it'll work anyways, but otherwise it times out and cancels the print. I am on the latest official firmware (2.7.2-2.3.2)

**Reply by xlordxcheater** [reply]
Same here. A5M, connected via WiFi. The uploading starts but no progress and after +-10secs, the error message pops out. Im in the latest firmware and latest orcaslicer version.

**Reply by masonare123** [reply]
same here adventure 5m

**Comment by N3dFlow** [comment]
Same here using a "Marlin" printer and Orca 2.2.0. I use MKS to connect, upload and print are fine but once the upload is finished, I still get the error "Error uploading to print host: Operation timed out"

![orca host error](https://github.com/user-attachments/assets/5ab054a0-f8a0-4096-895f-abb6a9d87009)

**Comment by joshuajohncohen** [comment]
Same issue

**Comment by doron007** [comment]
Same issue, though I observed this is only with larger files. Small file would upload successfully.
Is there a config file where we can increase the timeout?

**Comment by ovakisan** [comment]
Same issue here, over ethernet or wifi, the same. I had that issue with the original Flashforge Slicer so it seems it is on the printer side

**Comment by AndiRudi** [comment]
I have the same issue. The printer seems to react. It shows "Send the file: 0%" then quickly jumps to "Send the file: 100%".

**Comment by Hevdelurn** [comment]
(Edit: After testing more files, it seems to work most of the time, still a lot better than none.)

 If you can send smaller files just fine and it's just larger ones that seem to fail, then I have solved this issue!

You need to adjust the "sending_interval" in your OrcaSlicer.conf file.
Close OrcaSlicer
Go to: C:\Users\<your name>\AppData\Roaming\OrcaSlicer
Open OrcaSlicer.conf in a text editor
Scroll down or search for "sending_interval"
Mine was set to 5, I changed it to 100 and it worked perfectly.
Save and close.
![image](https://github.com/user-attachments/assets/b4c3d28c-8de6-4292-aa94-12c64aa1f692)

**Reply by nCclay** [reply]
Thank you so much brotha! You are a life saver!!!

**Reply by randomjoh** [reply]
This did not work for me. I still got the same error.

**Comment by gilphilbert** [comment]
I had another issue, although I suspect it's related.

FlashForge printers have a command called "get control" that's required for some operations, including upload. Only one device can have control at any time. When Orca tries to upload gcode files, it will request control of the printer but in some cases control access can be denied - Orca doesn't seem to have a way to handle this error, so it appears as a timeout.

In my case, another application already had control of the printer so the printer was denying the request. When I closed the other application, the upload was successful.

I would imagine that increasing the `sending_interval` value allows Orca to request control a few more times before it times out, increasing the chance of getting a successful response.

**Comment by Hevdelurn** [comment]
It was also hit or miss for me after I tried it. I fear I may have posted
my “fix” before I tested it enough.

**Comment by gilphilbert** [comment]
Another thing I've discovered is that the FlashForge 5M has a maximum number of connections. Not simultaneous, but total number of connections it will accept. Once you reach that limit, you must restart the printer. I expect this is true for the Pro too.

If you send lots of print jobs over the network to the printer and then suddenly you're getting errors, turn the printer off and back on and it should start working again.

**Comment by dbaarda** [comment]
For those who have never had this work, there was a bug where the Adv3 (and maybe others?) requires an extra delay between sending the file and the `~M29` "SaveFile" command, which prevented printing directly from Orca from working. This was fixed in #8905, which is not yet included in the V2.3.0 Official Release, but is included in the V2.3.1 alpha releases. This fix means I can now directly print from Orca Slicer (yay!).

Looking at the code I note that there are no calls to the progress update callback during the upload, hence 0% progress until the upload is complete. This is something that should be easy to fix if anyone wants a nice easy fix to tackle. Have a look at this [method](https://github.com/SoftFever/OrcaSlicer/blob/4c3081d65409e210905acceefb0749177d60ced1/src/slic3r/Utils/Flashforge.cpp#L114) and just add the progress callback updates around [here](https://github.com/SoftFever/OrcaSlicer/blob/4c3081d65409e210905acceefb0749177d60ced1/src/slic3r/Utils/Flashforge.cpp#L157).
